File: practice/algorithms/implementation/flatland_space_stations/flatland_space_stations.py
#!/usr/bin/env python

n, m = map(int, raw_input().strip().split())
c = sorted(map(int, raw_input().strip().split()))

res = max(c[0], n - c[m - 1] - 1)
    
for POS in xrange(1, m):
    res = max(res, (c[POS] - c[POS - 1]) // 2)
print res

File: practice/algorithms/implementation/flatland_space_stations/flatland_space_stations.cpp
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int main()
{
    int n, m, res;
    cin >> n >> m;
    vector<int> c(m);
    for (int pos = 0; pos < m; ++pos)
    {
       cin >> c[pos];
    }
    
    sort(c.begin(), c.end());
    
    res = max(c[0], n - 1 - c[m - 1]);
    for (int pos = 1; pos < m; ++pos)
    {
        res = max(res, (c[pos] - c[pos - 1]) / 2);
    }
    
    cout << res << endl;
    
    return 0;
}

